### PR TITLE
fix(package): Handle packages with only one file, a package-info.java.

### LINKED ISSUE #748: single package-info.java are not print
Hello,

I have a single package-info.java, there are no other file in the same package.

Spoon build it correctly, but it never print the package-info file.
It could change the output program behaviour.

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #748

### CODE REVIEW

File: src/main/java/spoon/support/JavaOutputProcessor.java [hunk 1]
```diff
@@ -195,17 +175,58 @@ public void createJavaFile(CtType<?> element) {
...
+		} else if (nameElement instanceof CtPackage) {
+			createPackageFile((CtPackage) nameElement);
 		}
 		printer.reset();
 	}
 
+	private void createPackageFile(CtPackage pack) {
+		// create package directory
+		File packageDir;
+		if (pack.isUnnamedPackage()) {
```
Comment (tdurieux) [inline src/main/java/spoon/support/JavaOutputProcessor.java]: This code is duplicate of https://github.com/INRIA/spoon/pull/766/files#diff-291edfebee2e5667f55882f29dfec128R143.

File: src/test/java/spoon/test/pkg/PackageTest.java [hunk 2]
```diff
@@ -80,4 +83,20 @@ public void testAnnotationOnPackage() throws Exception {
...
+		launcher.setSourceOutputDirectory("./target/spooned/package");
+		launcher.getEnvironment().setCommentEnabled(true);
+		launcher.run();
+
+		final CtPackage aPackage = launcher.getFactory().Package().get("spoon.test.pkg.testclasses.internal");
+		assertEquals(1, aPackage.getAnnotations().size());
+		assertEquals(3, aPackage.getComments().size());
+		assertEquals(CtComment.CommentType.JAVADOC, aPackage.getComments().get(0).getCommentType());
+		assertEquals(CtComment.CommentType.BLOCK, aPackage.getComments().get(1).getCommentType());
+		assertEquals(CtComment.CommentType.INLINE, aPackage.getComments().get(2).getCommentType());
```
Comment (monperrus) [inline src/test/java/spoon/test/pkg/PackageTest.java]: assert that package-info has been correctly rewritten to disk

File: src/test/java/spoon/test/pkg/PackageTest.java [hunk 3]
```diff
@@ -43,7 +46,7 @@ public void testPackage() throws Exception {
 		Assert.assertEquals(packageInfoFile.getCanonicalPath(), ctPackage.getPosition().getFile().getCanonicalPath());
 		Assert.assertEquals(1, ctPackage.getPosition().getLine());
 		Assert.assertEquals(1, ctPackage.getAnnotations().size());
-		Assert.assertEquals("This is test\nJavaDoc.", ctPackage.getDocComment());
```
Comment (monperrus) [inline src/test/java/spoon/test/pkg/PackageTest.java]: keep this assert

Comment (GerardPaligot) [inline src/test/java/spoon/test/pkg/PackageTest.java]: See the PR of @tdurieux about comments: https://github.com/INRIA/spoon/pull/554/files#diff-072d4b8ceab9c4e090f675ba9342b6f9L2050

A work has been done to remove usage of this method.

File: src/test/java/spoon/test/pkg/PackageTest.java (hunk 2 above)
Comment (GerardPaligot) [inline src/test/java/spoon/test/pkg/PackageTest.java]: Done.

**Pull request merged**